src/views/List.jsx

```diff
@@ -1,15 +1,69 @@
+import { useState, useEffect } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	// state for input
+	const [input, setInput] = useState('');
+	const [searchData, setSearchData] = useState(data);
+	const [isValid, setIsValid] = useState(false);
+	const [searchLength, setSearchLength] = useState(1);
+	// Handle event
+	const handleInputChange = (e) => {
+		setInput(e.target.value);
+	};
+
+	const handleInputClear = () => {
+		setInput('');
+	};
+
+	useEffect(() => {
+		const searchRegex = new RegExp(input, 'i');
+		const filteredData = data.filter((item) => searchRegex.test(item.name));
+		setSearchData(filteredData);
+	}, [data, input]);
+
+	useEffect(() => {
+		if (searchData.length > 0) {
+			setIsValid(true);
+			setSearchLength(50);
+		} else {
+			setIsValid(false);
+			setSearchLength(0);
+		}
+	}, [searchData, data]);
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
-			<ul>
-				{data &&
-					data.map((item, index) => <ListItem key={index} name={item.name} />)}
-			</ul>
+
+			<label htmlFor="input">
+				{' '}
+				Search for your item:
+				<input
+					type="text"
+					id="input"
+					value={input}
+					onChange={handleInputChange}
+					maxLength={searchLength}
+				/>
+			</label>
+			{input.length > 0 ? (
+				<button onClick={handleInputClear} value={input}>
```

Comment:
This value isn't necessary since you're just clearing the form, you can just use `button type="reset"` to clear the form without having to use the handleInputClear. I believe you'll need to have a `<form>` parent element but that way you can avoid overengineering a solution and having to rely on purely React for everything. :)

Reply:
I have been trying to fix this per your instructions. This is the code as I presently have it written: 

                            <form id="inputForm">
				Search for your item:
				<input
					type="text"
					id="inputField"
					value={input}
					onChange={handleInputChange}
					maxLength={searchLength}
				/>
				{input.length > 0 && (
					<button type="reset">x</button>
				)}
			    </form>
It does not work as intended, the button does nothing when clicked. I've tried multiple approaches like adding a `label`, including an `id` to the `button`, removing the `id` from `input` etc. but it will not work. For testing and demonstration purposes I added the following code to the bottom of the JSX. 

			<form id="testForm">
				<input type="text" id="test" />
				{input.length > 0 && (
					<button type="reset">x</button>
				)}
			</form>
And it works exactly as intended. What am I not seeing here?